components/skus/browser/skus_sdk_fetcher_impl.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/browser/skus_sdk_fetcher_impl.h"
+
+#include <utility>
+#include <vector>
+
+#include "brave/components/skus/browser/rs/cxx/src/lib.rs.h"
+#include "net/base/load_flags.h"
+#include "url/gurl.h"
+
+namespace {
+
+// Maximum accepted size of response from SKU SDK server. 1MB.
+const int kMaxResponseSize = 1024 * 1024;
+
+}  // namespace
+
+namespace skus {
+
+SkusSdkFetcherImpl::SkusSdkFetcherImpl(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(url_loader_factory) {}
+
+SkusSdkFetcherImpl::~SkusSdkFetcherImpl() {}
+
+void SkusSdkFetcherImpl::BeginFetch(
+    const skus::HttpRequest& req,
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<skus::HttpRoundtripContext>,
+             skus::HttpResponse)> callback,
+    rust::cxxbridge1::Box<skus::HttpRoundtripContext> ctx) {
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(static_cast<std::string>(req.url));
+  resource_request->method = static_cast<std::string>(req.method).c_str();
```

Comment:
why `c_str()`? Also why the static cast?

Reply:
Removed `c_str` in https://github.com/brave/brave-core/pull/10847/commits/529737f5e820c19972f937a4fdfb296afe572f8d. There were some weird things before you helped where the type had to be forced.  Static cast still required though as the values are `::rust::String` (which does have a proper std::string operator overload)